### PR TITLE
Allow one folder

### PR DESCRIPTION
**Pull request by renchauret**
if user zips a folder with csv and images, that's fine

### CODE REVIEW

File: client/src/components/BulkCreateForm/BulkCreateForm.js [hunk 1]
```diff
@@ -136,7 +135,8 @@ export default class BulkCreateForm extends React.Component {
                 const image = fileUpload.images[i]
                 if (!image.name.endsWith('.png') && !image.name.endsWith('.jpg')) {
                     if (image.name.endsWith('/')) {
-                        fileUpload.errorMessages.push('Files must be zipped directly. Do not include folders in your zip file.');
+                        continue
+                        // fileUpload.errorMessages.push('Files must be zipped directly. Do not include folders in your zip file.');
```
Comment (wrs1492) [inline client/src/components/BulkCreateForm/BulkCreateForm.js:139]: left commented code?

File: src/main/java/com/monumental/util/csvparsing/ZipFileHelper.java [hunk 2]
```diff
@@ -67,7 +67,16 @@ public static File convertZipEntryToFile(ZipFile zipFile, ZipEntry zipEntry) thr
...
-        String pathName = tempDirectoryPath + "/" + zipEntry.getName();
+        String[] nameArray = zipEntry.getName().split("/");
+        String name = nameArray[nameArray.length - 1];
+
+        String pathName = tempDirectoryPath;
+        if (zipEntry.getName().startsWith("/") || zipEntry.getName().startsWith("\\")
+                || tempDirectoryPath.endsWith("/") || tempDirectoryPath.endsWith("\\")) {
+            pathName += name;
+        } else {
+        pathName += "/" + name;
```
Comment (wrs1492) [inline src/main/java/com/monumental/util/csvparsing/ZipFileHelper.java:78]: indentation

**Review by wrs1492 — COMMENTED** [review]
Besides other comments, there where some other small problems,
1. The folder is counted as an image in the uploaded photos section. I'm not sure how the backend handles extra images, but it could cause a problem if it doesn't throw them out. It could be removed from the listing, but not a big deal if it isn't
2.  the error checking isn't great for the folders. No detection on if there are multiple folders, or if there are nested folders.
3.  How is the image name read/saved? I think the path is just appended right now, which is probably not ideal to save.

**Comment by renchauret** [comment]
After further testing, @wrs1492's other comments all work fine as is

**Review by wrs1492 — APPROVED** [review]

**Pull request merged**